pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -137,10 +137,18 @@ public CompletableFuture<Void> deleteTopicPoliciesAsync(TopicName topicName) {
             return CompletableFuture.completedFuture(null);
         }
         TopicName changeEvents = NamespaceEventsSystemTopicFactory.getEventsTopicName(topicName.getNamespaceObject());
-        return pulsarService.getNamespaceService().checkTopicExists(changeEvents).thenCompose(topicExistsInfo -> {
+        CompletableFuture<Boolean> changeEventTopicExists = pulsarService.getPulsarResources().getTopicResources()
+                .persistentTopicExists(changeEvents).thenCompose(nonPartitionedExists -> {
+            if (!nonPartitionedExists) {
+                return pulsarService.getPulsarResources().getTopicResources()
```

Comment:
Why do we need to double-check partition-0 here?

Reply:
- When users use global ZK, the partitioned topic `admin/partitioned-topic` ZK node was relied on both clusters.

- After users remove `local cluster` from the namespace level policies, the node `/managed-ledger/<topic>-partition-x` will be deleted by `PersistentTopic.checkReplication`, but the ZK node named `admin/partitioned-topic/<topic>` will not be removed because the remote cluster still uses it.

Uses the expression whether `partition-0` was deleted  is better than `partitoned metedata`